Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -65,126 +178,26 @@ void addPoint(float ptX, float ptY) {
-            case kCGPathElementMoveToPoint:
-            case kCGPathElementAddLineToPoint:
-                bbox.addPoint(_elements[i].points[0].x, _elements[i].points[0].y);
-                break;
-
-            case kCGPathElementAddCurveToPoint:
-                bbox.addPoint(_elements[i].points[0].x, _elements[i].points[0].y);
-                bbox.addPoint(_elements[i].points[1].x, _elements[i].points[1].y);
-                bbox.addPoint(_elements[i].points[2].x, _elements[i].points[2].y);
-                break;
-
-            case kCGPathElementAddQuadCurveToPoint:
-                bbox.addPoint(_elements[i].points[0].x, _elements[i].points[0].y);
-                bbox.addPoint(_elements[i].points[1].x, _elements[i].points[1].y);
-                break;
+/**
+ @Status Interoperable
+*/
+CGMutablePathRef CGPathCreateMutable() {
+    size_t memSize = sizeof(__CGPath) - sizeof(CFRuntimeBase);
+    CGPathRef ret = ((CGPathRef)(_CFRuntimeCreateInstance(kCFAllocatorDefault, CGPathGetTypeID(), memSize, NULL)));
+    __CGPath* mutableRet = const_cast<__CGPath*>(ret);
 
-            case kCGPathElementCloseSubpath:
-                break;
+    ComPtr<ID2D1Factory> factory = _CreateD2DFactoryInstance();
 
-            default:
-                assert(false);
-        }
+    if (FAILED(factory->CreatePathGeometry(&mutableRet->_pathGeometry))) {
+        FAIL_FAST();
     }
-
-    rectOut->origin.x = bbox.x;
-    rectOut->origin.y = bbox.y;
-    rectOut->size.width = bbox.x1 - bbox.x;
-    rectOut->size.height = bbox.y1 - bbox.y;
-}
-
-void _CGPathApplyPath(CGPathRef pathref, CGContextRef context) {
-    pathref->_applyPath(context);
-}
-
-// Not sure why this is necessary in addition to the public version, need to dig further
-void _CGPathGetBoundingBoxInternal(CGPathRef pathref, CGRect* rectOut) {
-    pathref->_getBoundingBox(rectOut);
-}
-
-void _CGPathAddElement(
-    CGPathRef path, CGPathElementType type, CGPoint p0 = CGPointZero, CGPoint p1 = CGPointZero, CGPoint p2 = CGPointZero) {
-    if (path->_count + 1 >= path->_max) {
-        path->_max += 32;
-        path->_elements = (CGPathElementInternal*)IwRealloc(path->_elements, path->_max * sizeof(CGPathElementInternal));
-        // Re-init all existing elements
-        for (int i = 0; i < path->_count; i++) {
-            CGPathElementInternal* element = &path->_elements[i];
-            element->init();
-        }
+    if (FAILED(mutableRet->_pathGeometry->Open(&mutableRet->_geometrySink))) {
+        FAIL_FAST();
     }
-    CGPathElementInternal* element = &path->_elements[path->_count];
-    // The new element needs to call init
-    // because it was created with alloc
-    element->init();
-    element->type = type;
-    element->points[0] = p0;
-    element->points[1] = p1;
-    element->points[2] = p2;
-
-    path->_count++;
-}
 
-/**
- @Status Interoperable
-*/
-CGMutablePathRef CGPathCreateMutable() {
-    return __CGPath::alloc(nil);
+    factory.~ComPtr();
```

Comment:
Additionally, using a value after it has been destructed is totally in the realm of undefined behaviour. It is the proverbial kiss of death.